api/client/client_test.go

```diff
@@ -210,3 +289,39 @@ func TestWaitForConnectionReady(t *testing.T) {
 	require.NoError(t, clt.GetConnection().Close())
 	require.Error(t, clt.waitForConnectionReady(ctx))
 }
+
+func TestLimitExceeded(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	addr := "localhost:6025"
+	startMockServer(t, addr)
```

Comment:
Nit: We should be moving away from binding explicit ports wherever possible.  You can have the OS assign you a random free port by binding port `0` and then extract the assigned port by checking the address of the returned listener.

Looks like `startMockServer` creates the listener on line 1, so adding the ability to return the bound addr should be a pretty simple tweak, and it'll be good for the long-term health of our tests.